src/main/java/frc/robot/commands/RotateVectorDriveCommand.java

```diff
@@ -57,10 +65,17 @@ public void execute() {
     double y = translationYSupplier.getAsDouble();
     double rotX = rotationXSupplier.getAsDouble();
     double rotY = rotationYSupplier.getAsDouble();
+    boolean isRobotRelative = isRobotRelativeSupplier.getAsBoolean();
 
     // if stick magnitude is greater then rotate angle mag
-    if (Util.vectorMagnitude(rotX, rotY) > Drive.ROTATE_VECTOR_MAGNITUDE) {
-      angle = Util.angleSnap(Util.vectorToAngle(-rotX, -rotY), angles);
+
+    double targetAngle = Util.angleSnap(Util.vectorToAngle(-rotX, -rotY), angles);
+
+    if (Util.vectorMagnitude(rotX, rotY) > Drive.ROTATE_VECTOR_MAGNITUDE && !isRobotRelative) {
+      angle = targetAngle;
+    } else if (Util.vectorMagnitude(rotX, rotY) > Drive.ROTATE_VECTOR_MAGNITUDE
+        && isRobotRelative) {
```

Comment:
ths boolean logic is reduntent and fairly ugly -- it could be expressed with a single if and a ternary.
`` if (Util.vectorMagnitude(rotX, rotY) > Drive.ROTATE_VECTOR_MAGNITUDE) ``
and then ``angle = isRobotRelative ? /** ect ect */``

Reply:
yeah that looks better